service/worker/workerdeployment/workflow.go

```diff
@@ -338,19 +358,37 @@ func (d *WorkflowRunner) handleSetCurrent(ctx workflow.Context, args *deployment
 		}
 	}
 
-	// tell new current that it's current
-	currUpdateArgs := &deploymentspb.SyncVersionStateUpdateArgs{
-		RoutingUpdateTime: updateTime,
-		CurrentSinceTime:  updateTime,
-		RampingSinceTime:  nil, // remove ramp if it existed
-		RampPercentage:    0,   // remove ramp if it existed
-	}
-	if _, err := d.syncVersion(ctx, newCurrentVersion, currUpdateArgs); err != nil {
-		return nil, err
+	if newCurrentVersion != worker_versioning.UnversionedVersionId {
+		// Tell new current version that it's current
+		currUpdateArgs := &deploymentspb.SyncVersionStateUpdateArgs{
+			RoutingUpdateTime: updateTime,
+			CurrentSinceTime:  updateTime,
+			RampingSinceTime:  nil, // remove ramp for that version if it was ramping
+			RampPercentage:    0,   // remove ramp for that version if it was ramping
+		}
+		if _, err := d.syncVersion(ctx, newCurrentVersion, currUpdateArgs); err != nil {
+			return nil, err
+		}
+	} else if d.State.RoutingConfig.RampingVersion == worker_versioning.UnversionedVersionId {
+		// If the new current is unversioned, and it was previously ramping, we need to tell
+		// all the task queues with unversioned ramp that they no longer have unversioned ramp.
+		// The task queues with unversioned ramp are the task queues of the previous current version.
+		// TODO (Carly): Should we ban people from changing the task queues in the current version while they have an unversioned ramp?
```

Comment:
discuss

Reply:
reasoning for banning this is that we use the current version's task queues to know what task queues the unversioned ramp is set on. We need that list to be able to unset the unversioned ramp.

Reply:
task queues can only be added to the current version, right? so the list will only get larger so you'd still unset all the unversioned ramps from the TQs who got the unversioned ramp.

Reply:
If someone does ignoreMissingTaskQueues = true then they can do whatever they want though right? including removing task queues from the current version?

Reply:
We would be unable to tell those task queues "you are done with unversioned ramp"

Reply:
Other option is to maintain a separate list inside the DeploymentLocalState of the task queues that we set unversioned ramp on ...